testsuite/src/test/java/org/ironjacamar/core/connectionmanager/pool/FlushCriTestCase.java

```diff
@@ -0,0 +1,352 @@
+import org.ironjacamar.rars.ResourceAdapterFactory;
+import org.ironjacamar.rars.security.UnifiedSecurityConnection;
+import org.ironjacamar.rars.security.UnifiedSecurityConnectionFactory;
+import org.ironjacamar.util.TestUtils;
+
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.resource.spi.TransactionSupport;
+
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * CRI test case w/ Prefill
+ *
+ * @author <a href="mailto:stefano.maestri@ironjacamar.org">Stefano Maestri</a>
+ */
+@RunWith(IronJacamar.class)
+@Configuration(full = true)
+@PreCondition(condition = AllChecks.class)
+@PostCondition(condition = AllChecks.class)
+public class FlushCriTestCase
+{
+   /**
+    * The connection factory w/o Tx
+    */
+   @Resource(mappedName = "java:/eis/UnifiedSecurityNoTxConnectionFactory")
+   private UnifiedSecurityConnectionFactory noTxCf;
+
+   /**
+    * The noTxCf w/o Tx w/o Prefill
+    */
+   @Resource(mappedName = "java:/eis/UnifiedSecurityNoTxNoPrefillConnectionFactory")
+   private UnifiedSecurityConnectionFactory noTxNoPrefillCf;
+
+
+   /**
+    * The deployment repository
+    */
+   @Inject
+   private static DeploymentRepository dr;
+
+   /**
+    * The resource adapter
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 1)
+   private ResourceAdapterArchive createResourceAdapter() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityRar();
+   }
+
+
+
+   /**
+    * The activation W/o Tx
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 2)
+   private ResourceAdaptersDescriptor createNoTxActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityDeployment(null, null,
+            TransactionSupport.TransactionSupportLevel.NoTransaction,
+            "UnifiedSecurityNoTxConnectionFactory", true, 10);
+   }
+
+
+   /**
+    * The activation w7o Tx w/o Prefill
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 3)
+   private ResourceAdaptersDescriptor createNoTxNoPrefillActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createUnifiedSecurityDeployment(null, null,
+            TransactionSupport.TransactionSupportLevel.NoTransaction,
+            "UnifiedSecurityNoTxNoPrefillConnectionFactory", false, 0);
+   }
+
+   /**
+    * test w/o Tx, 2 calls w/ same credentials
+    *
+    * @throws Throwable In case of an error
+    */
+   @Test
+   public void testPrefillInitialSize() throws Throwable
```

Comment:
Tests should be done by setting a flush strategy on the pool, and calling UnifiedSecurityConnection.fail(), then verify the result of listeners, and the individual states of each CL